src/barnetilsyn/BarnetilsynApp.tsx

```diff
@@ -43,45 +38,22 @@ const BarnetilsynApp = () => {
     verifiserAtBrukerErAutentisert(settAutentisering);
   }, [autentisert]);
 
-  const fetchPersonData = () => {
-    return hentPersonData()
-      .then((response) => {
-        settPerson({
-          type: PersonActionTypes.HENT_PERSON,
-          payload: response,
-        });
-        oppdaterSøknadMedBarn(response, response.barn);
-      })
-      .catch((e) => {
-        const feil = e.response?.data?.feil;
-
-        if (feil === 'adressesperre') {
-          logAdressesperre(ESkjemanavn.Barnetilsyn);
-          settAlvorlighetsgrad(EAlvorlighetsgrad.INFO);
-          settFeilmelding(
-            intl.formatMessage({
-              id: 'barnasbosted.feilmelding.adressebeskyttelse',
-            })
-          );
-        } else {
-          settFeilmelding(feil);
-        }
-
-        settError(true);
-      });
-  };
-
-  const oppdaterSøknadMedBarn = (person: IPerson, barneliste: IBarn[]) => {
-    const barnMedLabels = oppdaterBarnMedLabel(barneliste, intl);
+  const oppdaterSøknadMedBarn = (
+    person: PersonData,
+    barneliste: Barn[] | IBarn[]
+  ) => {
+    const barnMedLabels = oppdaterBarnMedLabel(barneliste as IBarn[], intl);
 
     settSøknad((prevSøknad) => {
       const prevBarn = prevSøknad.person.barn;
 
       const oppdatertBarn = [...prevBarn, ...barnMedLabels];
-
+      const sortertBarnelistePåMedforelder = oppdatertBarn.sort((a, b) =>
```

Comment:
Denne kodesnutten sorterer også selve variabelen `oppdatertBarn`. For å unngå potensielle overraskende side-effekter ville jeg derfor gjort denne linjen funksjonell slik:

`[...oppdatertBarn].sort`

Da vil ikke oppdatertBarn bli sortert. I tillegg er variabelen `a` ikke i bruk, så den ville jeg omdøpt til `_`

Reply:
Men linje 50 håper jeg oppretter en helt ny array? Slik at sorteringen på den er helt ufarlig? Eller sorteres faktisk da også prevBarn og barnMedLabels?

Reply:
Skjønner. Godt innspill!
Da tenker jeg at vi går for å fjerne variabelen `oppdatertBarn` og sorterer med en gang.

Reply:
[Fjernet oppdatertBarn](https://github.com/navikt/familie-ef-soknad/pull/1577/commits/dbd55c4e57c76aeef11f73d6dd83dfedf93f6df0).